tachyon/crypto/commitments/vector_commitment_scheme.h

```diff
@@ -62,33 +78,44 @@ class VectorCommitmentScheme {
     return derived->DoUnsafeSetup(size, params);
   }
 
-  // Commit to |container| and populates |result| with the commitment.
+  // Commit to |container| and populates |result| with the commitment if
+  // |batch_mode| is false. Otherwise, it stores the commitment in
+  // |batch_commitments_|.
   // Return false if the size of |container| doesn't match with the size of
   // parameters.
-  template <typename Container>
-  [[nodiscard]] bool Commit(const Container& container,
-                            Commitment* result) const {
-    const Derived* derived = static_cast<const Derived*>(this);
+  template <typename T = Derived, typename Container>
+  [[nodiscard]] bool Commit(const Container& container, Commitment* result) {
+    Derived* derived = static_cast<Derived*>(this);
+    if constexpr (VectorCommitmentSchemeTraits<T>::kSupportsBatchMode) {
+      if (batch_commitment_state_.batch_mode)
+        return derived->DoCommit(container, batch_commitment_state_);
+    }
     return derived->DoCommit(container, result);
   }
 
   // Commit to |container| with a |random_value| and populates |result| with the
-  // commitment. Return false if the size of |container| doesn't match with the
+  // commitment if |batch_mode| is false. Otherwise, it stores the commitment in
+  // |batch_commitments_|.
+  //. Return false if the size of |container| doesn't match with the
   // size of parameters.
-  template <typename Container>
+  template <typename T = Derived, typename Container>
   [[nodiscard]] bool Commit(const Container& container,
-                            const Field& random_value,
-                            Commitment* result) const {
-    const Derived* derived = static_cast<const Derived*>(this);
+                            const Field& random_value, Commitment* result) {
+    Derived* derived = static_cast<Derived*>(this);
+    if constexpr (VectorCommitmentSchemeTraits<T>::kSupportsBatchMode) {
+      if (batch_commitment_state_.batch_mode)
+        return derived->DoCommit(container, random_value,
+                                 batch_commitment_state_);
+    }
     return derived->DoCommit(container, random_value, result);
   }
 
   // Create an opening proof that proves that |members| belong to a
   // commitment.
   template <typename Container, typename Proof>
   [[nodiscard]] bool CreateOpeningProof(const Container& members,
-                                        Proof* proof) const {
-    const Derived* derived = static_cast<const Derived*>(this);
+                                        Proof* proof) {
+    Derived* derived = static_cast<Derived*>(this);
```

Comment:
Why did you remove const here?

Reply:
Since the `Commit` in univariate_polynomial_commitment_scheme.h is not marked as `const`, `DoCreateOpeningProof` in shplonk.h cannot be marked as `const` because it calls `this->Commit` inside the function. So I removed `const` here.